virtcontainers/pkg/oci/utils.go

```diff
@@ -75,23 +75,23 @@ const (
 
 const KernelModulesSeparator = ";"
 
-// CompatOCIProcess is a structure inheriting from spec.Process defined
+// CompatOCIProcess is a structure inheriting from specs.Process defined
 // in runtime-spec/specs-go package. The goal is to be compatible with
 // both v1.0.0-rc4 and v1.0.0-rc5 since the latter introduced a change
 // about the type of the Capabilities field.
 // Refer to: https://github.com/opencontainers/runtime-spec/commit/37391fb
 type CompatOCIProcess struct {
-	spec.Process
+	specs.Process
 	Capabilities interface{} `json:"capabilities,omitempty" platform:"linux"` //nolint:govet
 }
 
-// CompatOCISpec is a structure inheriting from spec.Spec defined
+// CompatOCISpec is a structure inheriting from specs.Spec defined
 // in runtime-spec/specs-go package. It relies on the CompatOCIProcess
 // structure declared above, in order to be compatible with both
 // v1.0.0-rc4 and v1.0.0-rc5.
 // Refer to: https://github.com/opencontainers/runtime-spec/commit/37391fb
 type CompatOCISpec struct {
-	spec.Spec
+	specs.Spec
```

Comment:
I *think* this `struct` can be made private now? (`compatOCISpec`).

Note that there is a comment in `virtcontainers/persist/api/container.go` about it still though:

```go
// BundlePath saves container OCI config.json, which can be unmarshaled
// and translated to "CompatOCISpec"
BundlePath string
```

Is that comment still valid?

Reply:
The comment is still valid, besides there will be another PR about persist data doing the config file refactoring.

And for the private change, I'll need some more investigation.

Reply:
Thanks @WeiZhang555! And thanks for tackling the overall issue on this PR - great refactor! ;)

Reply:
I checked the code and made the "CompatOCISpec" private structure, same for "ContainerCapabilities()". Now they're "compatOCISpec" and "containerCapabilities()".

Thanks @jodh-intel for your great suggestion!